In [33]:
import pandas as pd

In [50]:
data_df: pd.DataFrame = pd.read_sas('/content/LLCP2023.XPT', iterator=False, chunksize=None)

In [51]:
data_df.head()

_STATE  FMONTH        IDATE IMONTH   IDAY    IYEAR  DISPCODE  \
0     1.0     1.0  b'03012023'  b'03'  b'01'  b'2023'    1100.0   
1     1.0     1.0  b'01062023'  b'01'  b'06'  b'2023'    1100.0   
2     1.0     1.0  b'03082023'  b'03'  b'08'  b'2023'    1100.0   
3     1.0     1.0  b'03062023'  b'03'  b'06'  b'2023'    1100.0   
4     1.0     1.0  b'01062023'  b'01'  b'06'  b'2023'    1100.0   

           SEQNO          _PSU  CTELENM1  ...      DROCDY4_  _RFBING6  \
0  b'2023000001'  2.023000e+09       1.0  ...  5.397605e-79       1.0   
1  b'2023000002'  2.023000e+09       1.0  ...  5.397605e-79       1.0   
2  b'2023000003'  2.023000e+09       1.0  ...  5.397605e-79       1.0   
3  b'2023000004'  2.023000e+09       1.0  ...  5.397605e-79       1.0   
4  b'2023000005'  2.023000e+09       1.0  ...  7.000000e+00       1.0   

       _DRNKWK2  _RFDRHV8  _FLSHOT7  _PNEUMO3  _AIDTST4  _RFSEAT2  _RFSEAT3  \
0  5.397605e-79       1.0       2.0       2.0       2.0       1.0       1.0   
1  5.397605e-79       1.0       1.0       1.0       2.0       1.0       1.0   
2  5.397605e-79       1.0       1.0       1.0       2.0       1.0       1.0   
3  5.397605e-79       1.0       1.0       1.0       1.0       1.0       1.0   
4  4.700000e+01       1.0       2.0       1.0       2.0       1.0       1.0   

   _DRNKDRV  
0       9.0  
1       9.0  
2       9.0  
3       9.0  
4       2.0  

[5 rows x 350 columns]

In [52]:
data_df.shape

(433323, 350)

In [53]:
data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 433323 entries, 0 to 433322
Columns: 350 entries, _STATE to _DRNKDRV
dtypes: float64(345), object(5)
memory usage: 1.1+ GB


In [54]:
df = data_df[['_AGEG5YR', '_ASTHMS1', '_BMI5CAT', '_CHLDCNT', '_DRDXAR2', '_EDUCAG', '_IMPRACE', '_INCOMG1', '_MENT14D', '_MICHD', '_PHYS14D', '_RFCHOL3', '_RFHYPE6', '_SMOKER3', 'ADDEPEV3', 'CHCCOPD3', 'CHCKDNY2', 'CHCOCNC1', 'CHCSCNC1', 'CVDSTRK3', 'DIABETE4', 'GENHLTH', 'POORHLTH', 'SDHSTRE1', 'SDLONELY', 'SEXVAR', '_DRNKWK2', '_TOTINDA']]


In [75]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 433323 entries, 0 to 433322
Data columns (total 28 columns):
 #   Column    Non-Null Count   Dtype  
---  ------    --------------   -----  
 0   _AGEG5YR  433323 non-null  float64
 1   _ASTHMS1  433323 non-null  float64
 2   _BMI5CAT  392788 non-null  float64
 3   _CHLDCNT  433323 non-null  float64
 4   _DRDXAR2  430763 non-null  float64
 5   _EDUCAG   433323 non-null  float64
 6   _IMPRACE  433323 non-null  float64
 7   _INCOMG1  433323 non-null  float64
 8   _MENT14D  433323 non-null  float64
 9   _MICHD    428738 non-null  float64
 10  _PHYS14D  433323 non-null  float64
 11  _RFCHOL3  381512 non-null  float64
 12  _RFHYPE6  433323 non-null  float64
 13  _SMOKER3  433323 non-null  float64
 14  ADDEPEV3  433320 non-null  float64
 15  CHCCOPD3  433320 non-null  float64
 16  CHCKDNY2  433320 non-null  float64
 17  CHCOCNC1  433320 non-null  float64
 18  CHCSCNC1  433320 non-null  float64
 19  CVDSTRK3  433319 non-null  float64
 20  DIAB

Dataset has 433 323 observations and 350 columns in total

Dropping not needed columns:

In [ ]:
columns_to_drop = [
    '_STATE', # State FIPS Code
    'FMONTH', # File Month
    'IDATE', # Interview Date
    'IMONTH', # Interview Month
    'IDAY', # Interview Day
    'IYEAR', # Interview Year
    'DISPCODE', # Final Disposition (interview completed fully of partially)
    'SEQNO', # Annual Sequence Number
    '_PSU', # Primary Sampling Unit (Equal to Annual Sequence Number)
    'CTELENM1', # Correct telephone number?
    'PVTRESD1', # Is this a private residence?
    'COLGHOUS', # Do you live in college housing?
    'STATERE1', # Do you currently live in  ____(state)____?
    'CELPHON1', # Is this a cell telephone?
    'LADULT1', # Are you 18 years of age or older? [NA: 344,978, Yes: 88,212, No: 133]
    'NUMADULT', # how many members of your household, including yourself, are 18 years of age or older? [NA: 344,993]
    'RESPSLC1', # The person in your household that I need to speak with...
    'LANDSEX2',
    'LNDSXBRT',
    'SAFETIME', # Is this a safe time to talk with you?
    'CTELNUM1',
    'CELLFON5', # Is this a cell phone?
    'CADULT1', # Are you 18 years of age or older?
    'CELLSEX2',
    'CELSXBRT',
    'PVTRESD3',
    'CCLGHOUS',
    'CSTATE1',
    'LANDLINE',
    'HHADULT',
    'PRIMINS1', # What is the current source of your primary health insurance? (irrelevant)
    'NUMHHOL4', # Number of landline telephones
    'NUMPHON4',
    'FLSHTMY3', # During what month and year did you receive your most recent flu vaccine
    '_STSTR',
    '_STRWT',
    '_DUALUSE',
    '_SEX', # Very close to SEXVAR variable
    '_METSTAT',
    '_URBSTAT',
    'MSCODE',
    '_WT2RAKE',
    '_LLCPWT2',
    '_RAWRAKE',
    '_CLLCPWT',
    '_DUALCOR',
    '_LLCPWT',
    'PADUR1_',
    'PADUR2_',
    'PAMIN13_',
    'PAMIN23_',
    'PA3MIN_',
    'PAVIG13_',
    'PAVIG23_',
    'PA3VIGM_',
    'HTIN4',
    'HTM4',
    'WTKG3',
    '_BMI5'
]
data_df.drop(columns=columns_to_drop, inplace=True)

In [ ]:
data_df.shape

(433323, 291)

34 columns were dropped

Drop columns that are consisted of NAs mostly:

In [ ]:
cols_na_counts: pd.Series = data_df.isna().sum()
cols_prim_na: pd.Series = cols_na_counts[cols_na_counts >= data_df.shape[0] * 0.6]
cols_prim_na

ASTHNOW     368957
DIABAGE4    373537
PREGNANT    357115
FALLINJ5    355584
SMOKDAY2    274684
             ...  
CASTHNO2    428574
_CRACE1     382113
CAGEG       391398
_FLSHOT7    262500
_PNEUMO3    262500
Length: 150, dtype: int64

In [ ]:
data_df.drop(columns=cols_prim_na.index, inplace=True)
data_df.shape

(433323, 141)

153 columns were dropped

Drop columns with too much skewness in data:

In [ ]:
# Made exceptions for columns in category "Chronic Health Conditions" or related
cols_to_exclude = [
    'CVDINFR4', # Ever Diagnosed with Heart Attack
    'CVDCRHD4', # Ever Diagnosed with Angina or Coronary Heart Disease
    'CVDSTRK3', # Ever Diagnosed with a Stroke
    'ASTHMA3', # Ever Told Had Asthma
    'CHCSCNC1', # (Ever told) (you had) skin cancer that is not melanoma?
    'CHCOCNC1', # (Ever told) (you had) melanoma or any other types of cancer?
    'CHCCOPD3', # (Ever told) (you had) C.O.P.D. (chronic obstructive pulmonary disease), emphysema or chronic bronchitis?
    'ADDEPEV3', # (Ever told) you had a depressive disorder
    'CHCKDNY2', # Ever told you have kidney disease?
    'HAVARTH4', # Told Had Arthritis
    'DIABETE4', # (Ever told) you had diabetes
    'EXRACT12', # What type of physical activity or exercise did you spend the most time doing during the past month?
    'DRNKANY6', # Drink any alcoholic beverages in past 30 days
    'EXERANY2'
]

In [ ]:
# Also exclude calculated variables, will decide for each of them separately
var_skewness = data_df.loc[:, ~data_df.columns.str.startswith('_')].drop(columns=cols_to_exclude).skew()
data_df.drop(columns=var_skewness[abs(var_skewness) > 1].index, inplace=True)

In [ ]:
data_df.columns

Index(['SEXVAR', 'GENHLTH', 'PHYSHLTH', 'MENTHLTH', 'POORHLTH', 'EXERANY2',
       'EXRACT12', 'EXRACT22', 'STRENGTH', 'BPHIGH6', 'CVDINFR4', 'CVDCRHD4',
       'CVDSTRK3', 'ASTHMA3', 'CHCSCNC1', 'CHCOCNC1', 'CHCCOPD3', 'ADDEPEV3',
       'CHCKDNY2', 'HAVARTH4', 'DIABETE4', 'EDUCA', 'EMPLOY1', 'ALCDAY4',
       'SDLONELY', 'SDHSTRE1', '_IMPRACE', '_CHISPNC', '_RFHLTH', '_PHYS14D',
       '_MENT14D', '_HLTHPL1', '_HCVU653', '_TOTINDA', 'METVL12_', 'METVL22_',
       'ACTIN13_', 'ACTIN23_', '_MINAC12', '_MINAC22', '_PACAT3', '_PAINDX3',
       '_PA150R4', '_PA300R4', '_PA30023', '_PASTRNG', '_PAREC3', '_PASTAE3',
       '_RFHYPE6', '_CHOLCH3', '_RFCHOL3', '_MICHD', '_LTASTH1', '_CASTHM1',
       '_ASTHMS1', '_DRDXAR2', '_MRACE1', '_HISPANC', '_RACE', '_RACEG21',
       '_RACEGR3', '_RACEPRV', '_AGEG5YR', '_AGE65YR', '_AGE80', '_AGE_G',
       '_BMI5CAT', '_RFBMI5', '_CHLDCNT', '_EDUCAG', '_INCOMG1', '_SMOKER3',
       '_RFSMOK3', '_CURECI2', 'DRNKANY6', '_RFBING6', '_DRNKWK2', '_RFDRHV8'

Choose between calculated and original columns, drop not needed ones:

In [ ]:
cols_with_replacement = [
    '_RFHLTH',
    '_PASTRNG',
    '_PAREC3',
    '_PASTAE3',
    'BPHIGH6',
    'CVDINFR4',
    'CVDCRHD4',
    'ASTHMA3',
    '_LTASTH1',
    '_CASTHM1',
    'HAVARTH4',
    'EDUCA',
    'ALCDAY4',
    '_CHISPNC',
    '_RFHLTH',
    '_HCVU653',
    'METVL12_',
    'METVL22_',
    '_PAREC3',
    '_PASTAE3',
    '_HISPANC',
    '_RACE',
    '_RACEG21',
    '_RACEGR3',
    '_RACEPRV',
    '_AGE65YR',
    '_AGE80',
    '_AGE_G',
    '_RFBMI5',
    '_RFSMOK3',
    '_CURECI2',
    '_RFSEAT2'
]
data_df.drop(columns=cols_with_replacement, inplace=True)

In [ ]:
data_df.columns

Index(['SEXVAR', 'GENHLTH', 'PHYSHLTH', 'MENTHLTH', 'POORHLTH', 'EXERANY2',
       'EXRACT12', 'EXRACT22', 'STRENGTH', 'CVDSTRK3', 'CHCSCNC1', 'CHCOCNC1',
       'CHCCOPD3', 'ADDEPEV3', 'CHCKDNY2', 'DIABETE4', 'EMPLOY1', 'SDLONELY',
       'SDHSTRE1', '_IMPRACE', '_PHYS14D', '_MENT14D', '_HLTHPL1', '_TOTINDA',
       'ACTIN13_', 'ACTIN23_', '_MINAC12', '_MINAC22', '_PACAT3', '_PAINDX3',
       '_PA150R4', '_PA300R4', '_PA30023', '_RFHYPE6', '_CHOLCH3', '_RFCHOL3',
       '_MICHD', '_ASTHMS1', '_DRDXAR2', '_MRACE1', '_AGEG5YR', '_BMI5CAT',
       '_CHLDCNT', '_EDUCAG', '_INCOMG1', '_SMOKER3', 'DRNKANY6', '_RFBING6',
       '_DRNKWK2', '_RFDRHV8', '_AIDTST4', '_RFSEAT3', '_DRNKDRV'],
      dtype='object')

Remaining columns and their descriptions:

- SEXVAR: Sex of Respondent
- _AGEG5YR: "Reported age in five-year age categories calculated variable"
- _IMPRACE: "Imputed race/ethnicity value"
- _MRACE1: "Calculated non-Hispanic Race including multiracial"
- _EDUCAG: "Computed level of education completed categories"
- EMPLOY1: "Employment Status"
- _INCOMG1: "Computed income categories"
- _HLTHPL1: "Have any health insurance"


Health
- GENHLTH: General health status
- PHYSHLTH: Number of Days Physical Health Not Good
- MENTHLTH: Number of Days Mental Health Not Good
- POORHLTH: "During the past 30 days, for about how many days did poor physical or mental health keep you from doing your usual activities, such as self-care, work, or recreation?"

Diseases
- CVDSTRK3: "Ever Diagnosed with a Stroke"
- CHCSCNC1: "(Ever told) (you had) skin cancer that is not melanoma?"
- CHCOCNC1: "(Ever told) (you had)  melanoma or any other types of cancer?"
- CHCCOPD3: "Ever told you had C.O.P.D. emphysema or chronic bronchitis?"
- CHCKDNY2: "Ever told you have kidney disease?"
- DIABETE4: "(Ever told) you had diabetes"
- _RFHYPE6: "High Blood Pressure Calculated Variable"
- _CHOLCH3: "Cholesterol Checked Calculated Variable"
- _RFCHOL3: "High Cholesterol Calculated Variable"
- _MICHD: "Ever had CHD or MI"
- _ASTHMS1: "Computed Asthma Status"
- _DRDXAR2: "Respondents diagnosed with arthritis"
- _BMI5CAT: "Computed body mass index categories"

Mental Health
- ADDEPEV3: "(Ever told) you had a depressive disorder"
- _MENT14D: "Computed Mental Health Status"
- SDLONELY: "How often do you feel lonely?"
- SDHSTRE1: "How often have you felt this kind of stress?"


Physical Activity
- EXERANY2: "Exercise in Past 30 Days"
- EXRACT12: "What type of physical activity or exercise did you spend the most time doing during the past month?"
- EXRACT22: "What other type of physical activity gave you the next most exercise during the past month?"
- STRENGTH: "How many times did you do physical activities or exercises to STRENGTHEN your muscles?"
- _PHYS14D: "Computed Physical Health Status"
- _TOTINDA: "Leisure Time Physical Activity Calculated Variable"
- ACTIN13_: "Estimated Activity Intensity for First Activity"
- ACTIN23_: "Estimated Activity Intensity for Second Activity"
- _MINAC12: "Minutes of Physical Activity per week for First Activity"
- _MINAC22: "Minutes of Physical Activity per week for Second Activity"
- _PACAT3: "Physical Activity Categories"
- _PAINDX3: "Physical Activity Index"
- _PA150R4: "150 Minute Physical Activity Calculated Variable"
- _PA300R4: "300 Minute Physical Activity Calculated Variable"
- _PA30023: "300 Minute Physical Activity 2-Level Calculated Variable"

Alcohol
- DRNKANY6: "Drink any alcoholic beverages in past 30 days"
- _RFBING6: "Binge Drinking Calculated Variable"
- _DRNKWK2: "Computed number of drinks of alcohol beverages per week"
- _DRNKDRV: "Drinking and Driving"
- _RFDRHV8: "Heavy Alcohol Consumption  Calculated Variable"

Other
- _SMOKER3: "Computed Smoking Status"
- _RFSEAT3: "Always Wear Seat Belts"
- _AIDTST4: "Ever been tested for HIV calculated variable"
- _CHLDCNT: "Computed number of children in household"







*   _IMPRACE: "Imputed race/ethnicity value" vs _MRACE1: "Calculated non-Hispanic Race including multiracial" - если надо много переменныех то второе, если хватит белый, черный и т.д., то первое
* MENTHLTH: Number of Days Mental Health Not Good vs _MENT14D: "Computed Mental Health Status" - первое это численное значение с плохими днями, второе это уже категорияльное с 0, 1-13 и 14+ плохих дней
* ASTHMA3: "Ever Told Had Asthma" vs _ASTHMS1: "Computed Asthma Status" - лучше взять только вторую
* CHCSCNC1: "(Ever told) (you had) skin cancer that is not melanoma?" vs CHCOCNC1: "(Ever told) (you had)  melanoma or any other types of cancer?" - их я бы объеденила
* _CHOLCH3: "Cholesterol Checked Calculated Variable" vs _RFCHOL3: "High Cholesterol Calculated Variable" - оставить только вторую

*   из всей физ активности оставить только EXERANY2: "Exercise in Past 30 Days", _PHYS14D: "Computed Physical Health Status", _PACAT3: "Physical Activity Categories"





# Lana

In [147]:
df_lana = df.copy()

In [110]:
df_lana.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 433323 entries, 0 to 433322
Data columns (total 28 columns):
 #   Column    Non-Null Count   Dtype  
---  ------    --------------   -----  
 0   _AGEG5YR  433323 non-null  float64
 1   _ASTHMS1  433323 non-null  float64
 2   _BMI5CAT  392788 non-null  float64
 3   _CHLDCNT  433323 non-null  float64
 4   _DRDXAR2  430763 non-null  float64
 5   _EDUCAG   433323 non-null  float64
 6   _IMPRACE  433323 non-null  float64
 7   _INCOMG1  433323 non-null  float64
 8   _MENT14D  433323 non-null  float64
 9   _MICHD    428738 non-null  float64
 10  _PHYS14D  433323 non-null  float64
 11  _RFCHOL3  381512 non-null  float64
 12  _RFHYPE6  433323 non-null  float64
 13  _SMOKER3  433323 non-null  float64
 14  ADDEPEV3  433320 non-null  float64
 15  CHCCOPD3  433320 non-null  float64
 16  CHCKDNY2  433320 non-null  float64
 17  CHCOCNC1  433320 non-null  float64
 18  CHCSCNC1  433320 non-null  float64
 19  CVDSTRK3  433319 non-null  float64
 20  DIAB

In [149]:
def label_Temp(row):
   if row['_DRNKWK2'] < 0.99:
      return 'No'
   else : return 'Yes'

df_lana.apply(label_Temp, axis=1)

0          No
1          No
2          No
3          No
4         Yes
         ... 
433318    Yes
433319     No
433320    Yes
433321     No
433322    Yes
Length: 433323, dtype: object

In [150]:
df_lana['_DRNKWK21'] = df_lana.apply(label_Temp, axis=1)
df_lana = df_lana.drop(['_DRNKWK2'], axis=1)

In [151]:
sex_col = {
          1: 'Male',
          2: 'Female'
          }
df_lana['SEXVAR'] = df_lana['SEXVAR'].replace(sex_col)

hlth_col = {
          1: 'Excellent',
          2: 'Very good',
          3: 'Good',
          4: 'Fair',
          5: 'Poor'
          }
df_lana['GENHLTH'] = df_lana['GENHLTH'].replace(hlth_col)

phlth_col = {
          88: 'None'
          }
df_lana['POORHLTH'] = df_lana['POORHLTH'].replace(phlth_col)

exer_col = {
          1: 'Yes',
          2: 'No'
          }
df_lana['CVDSTRK3'] = df_lana['CVDSTRK3'].replace(exer_col)
df_lana['CHCCOPD3'] = df_lana['CHCCOPD3'].replace(exer_col)
df_lana['ADDEPEV3'] = df_lana['ADDEPEV3'].replace(exer_col)
df_lana['CHCKDNY2'] = df_lana['CHCKDNY2'].replace(exer_col)
df_lana['_MICHD'] = df_lana['_MICHD'].replace(exer_col)
df_lana['_DRDXAR2'] = df_lana['_DRDXAR2'].replace(exer_col)
df_lana['_TOTINDA'] = df_lana['_TOTINDA'].replace(exer_col)
df_lana['CHCSCNC1'] = df_lana['CHCSCNC1'].replace(exer_col)
df_lana['CHCOCNC1'] = df_lana['CHCOCNC1'].replace(exer_col)


diabete_col = {
          1: 'Yes',
          2: 'Yes',
          3: 'No',
          4: 'Yes'
          }
df_lana['DIABETE4'] = df_lana['DIABETE4'].replace(diabete_col)

bmi_col = {
          1: '77',
          2: 'Normal Weight',
          3: 'Overweight',
          4: 'Obese'
          }
df_lana['_BMI5CAT'] = df_lana['_BMI5CAT'].replace(bmi_col)

sad_col = {
          1: 'Always',
          2: 'Usually',
          3: 'Sometimes',
          4: 'Rarely',
          5: 'Never'
          }
df_lana['SDLONELY'] = df_lana['SDLONELY'].replace(sad_col)
df_lana['SDHSTRE1'] = df_lana['SDHSTRE1'].replace(sad_col)

race_col = {
          1: 'White',
          2: 'Other',
          3: 'Other',
          4: 'Other',
          5: 'Other',
          6: 'Other'
          }
df_lana['_IMPRACE'] = df_lana['_IMPRACE'].replace(race_col)

phy_col = {
          1: 'Zero days',
          2: '1-13 days',
          3: '	14+ days'
          }
df_lana['_PHYS14D'] = df_lana['_PHYS14D'].replace(phy_col)
df_lana['_MENT14D'] = df_lana['_MENT14D'].replace(phy_col)

hbp_col = {
          1: 'No',
          2: 'Yes'
          }
df_lana['_RFHYPE6'] = df_lana['_RFHYPE6'].replace(hbp_col)
df_lana['_RFCHOL3'] = df_lana['_RFCHOL3'].replace(hbp_col)

ast_col = {
          1: 'Yes',
          2: 'Yes',
          3: 'No'
          }
df_lana['_ASTHMS1'] = df_lana['_ASTHMS1'].replace(ast_col)

age_col = {
          1: 'Age 18 to 29',
          2: 'Age 18 to 29',
          3: 'Age 30 to 39',
          4: 'Age 30 to 39',
          5: 'Age 40 to 49',
          6: 'Age 40 to 49',
          7: 'Age 50 to 59',
          8: 'Age 50 to 59',
          9: 'Age 60 to 69',
          10: 'Age 60 to 69',
          11: 'Age 70+',
          12: 'Age 70+',
          13: 'Age 70+',
          14: '7'
          }
df_lana['_AGEG5YR'] = df_lana['_AGEG5YR'].replace(age_col)

chld_col = {
          1: 'No',
          2: 'Yes',
          3: 'Yes',
          4: 'Yes',
          5: 'Yes',
          6: 'Yes'
          }
df_lana['_CHLDCNT'] = df_lana['_CHLDCNT'].replace(chld_col)

edu_col = {
          1: 'No',
          2: 'High School',
          3: 'Attended College',
          4: 'College or Technical School'
          }
df_lana['_EDUCAG'] = df_lana['_EDUCAG'].replace(edu_col)

smoker_col = {
          1: 'Yes',
          2: 'Yes',
          3: 'Yes',
          4: 'No'
          }
df_lana['_SMOKER3'] = df_lana['_SMOKER3'].replace(smoker_col)

smoker_col = {
          1: 'Yes',
          2: 'Yes',
          3: 'Yes',
          4: 'No'
          }
df_lana['_SMOKER3'] = df_lana['_SMOKER3'].replace(smoker_col)

age_col = {
          1: 'Less than $15,000',
          2: '$15,000 to < $25,000',
          3: '$25,000 to < $35,000',
          4: '$35,000 to < $50,000',
          5: '$50,000 to < $100,000',
          6: '$100,000 to < $200,00',
          7: '$200,000+',
          9: '77'
          }
df_lana['_INCOMG1'] = df_lana['_INCOMG1'].replace(age_col)

df_lana = df_lana[~df_lana[['GENHLTH', 'CVDSTRK3', 'CHCSCNC1', 'CHCOCNC1', 'CHCCOPD3', 'ADDEPEV3', 'CHCKDNY2', 'DIABETE4', 'SDLONELY', 'SDHSTRE1', '_PHYS14D', '_MENT14D', '_RFHYPE6', '_RFCHOL3', '_MICHD', '_ASTHMS1', '_AGEG5YR', '_CHLDCNT', '_EDUCAG', '_SMOKER3', '_TOTINDA']].isin([7, 9, '7']).any(axis=1)]
df_lana = df_lana[~df_lana[['POORHLTH', '_BMI5CAT', '_INCOMG1']].isin([77, 99, '77']).any(axis=1)]


In [152]:
import numpy as np
df_lana['CHCSCNC'] = np.where((df_lana['CHCSCNC1'] == 'Yes') | (df_lana['CHCOCNC1'] == 'Yes'), 'Yes', 'No')
df_lana = df_lana.drop(['CHCSCNC1', 'CHCOCNC1'], axis=1)

In [153]:
df_lana.info()

<class 'pandas.core.frame.DataFrame'>
Index: 305670 entries, 2 to 433321
Data columns (total 27 columns):
 #   Column     Non-Null Count   Dtype 
---  ------     --------------   ----- 
 0   _AGEG5YR   305670 non-null  object
 1   _ASTHMS1   305670 non-null  object
 2   _BMI5CAT   293244 non-null  object
 3   _CHLDCNT   305670 non-null  object
 4   _DRDXAR2   304652 non-null  object
 5   _EDUCAG    305670 non-null  object
 6   _IMPRACE   305670 non-null  object
 7   _INCOMG1   305670 non-null  object
 8   _MENT14D   305670 non-null  object
 9   _MICHD     303835 non-null  object
 10  _PHYS14D   305670 non-null  object
 11  _RFCHOL3   274425 non-null  object
 12  _RFHYPE6   305670 non-null  object
 13  _SMOKER3   305670 non-null  object
 14  ADDEPEV3   305670 non-null  object
 15  CHCCOPD3   305670 non-null  object
 16  CHCKDNY2   305670 non-null  object
 17  CVDSTRK3   305669 non-null  object
 18  DIABETE4   305670 non-null  object
 19  GENHLTH    305670 non-null  object
 20  POORHLTH 

In [162]:
for column in df_lana.select_dtypes(include='object').columns:
    print(df_lana[column].astype(str).value_counts().sort_index())
    print()

_AGEG5YR
Age 18 to 29     6504
Age 30 to 39    10455
Age 40 to 49    12041
Age 50 to 59    14402
Age 60 to 69    17223
Age 70+         18583
Name: count, dtype: int64

_ASTHMS1
No     64561
Yes    14647
Name: count, dtype: int64

_BMI5CAT
Normal Weight    20834
Obese            31257
Overweight       27117
Name: count, dtype: int64

_CHLDCNT
No     58030
Yes    21178
Name: count, dtype: int64

_DRDXAR2
No     47437
Yes    31771
Name: count, dtype: int64

_EDUCAG
Attended College               22264
College or Technical School    36665
High School                    16990
No                              3289
Name: count, dtype: int64

_IMPRACE
Other    19236
White    59972
Name: count, dtype: int64

_INCOMG1
$100,000 to < $200,00    17313
$15,000 to < $25,000      7362
$200,000+                 5766
$25,000 to < $35,000      8661
$35,000 to < $50,000     10702
$50,000 to < $100,000    24733
Less than $15,000         4671
Name: count, dtype: int64

_MENT14D
\t14+ days    17544
1-13 days 

In [159]:
df_lana.duplicated(keep='first').sum()

666

In [160]:
df_lana.drop_duplicates(keep='first', inplace=True)
df_lana.duplicated(keep='first').sum()

0

In [155]:
df_lana.isna().sum()

_AGEG5YR          0
_ASTHMS1          0
_BMI5CAT      12426
_CHLDCNT          0
_DRDXAR2       1018
_EDUCAG           0
_IMPRACE          0
_INCOMG1          0
_MENT14D          0
_MICHD         1835
_PHYS14D          0
_RFCHOL3      31245
_RFHYPE6          0
_SMOKER3          0
ADDEPEV3          0
CHCCOPD3          0
CHCKDNY2          0
CVDSTRK3          1
DIABETE4          0
GENHLTH           0
POORHLTH     130284
SDHSTRE1     140534
SDLONELY     138852
SEXVAR            0
_TOTINDA          0
_DRNKWK21         0
CHCSCNC           0
dtype: int64

In [156]:
df_lana.dropna(inplace=True)

In [161]:
df_lana.info()

<class 'pandas.core.frame.DataFrame'>
Index: 79208 entries, 2 to 431254
Data columns (total 27 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   _AGEG5YR   79208 non-null  object
 1   _ASTHMS1   79208 non-null  object
 2   _BMI5CAT   79208 non-null  object
 3   _CHLDCNT   79208 non-null  object
 4   _DRDXAR2   79208 non-null  object
 5   _EDUCAG    79208 non-null  object
 6   _IMPRACE   79208 non-null  object
 7   _INCOMG1   79208 non-null  object
 8   _MENT14D   79208 non-null  object
 9   _MICHD     79208 non-null  object
 10  _PHYS14D   79208 non-null  object
 11  _RFCHOL3   79208 non-null  object
 12  _RFHYPE6   79208 non-null  object
 13  _SMOKER3   79208 non-null  object
 14  ADDEPEV3   79208 non-null  object
 15  CHCCOPD3   79208 non-null  object
 16  CHCKDNY2   79208 non-null  object
 17  CVDSTRK3   79208 non-null  object
 18  DIABETE4   79208 non-null  object
 19  GENHLTH    79208 non-null  object
 20  POORHLTH   79208 non-null  objec

In [ ]:
for column in df_lana.select_dtypes(include='object').columns:
    print(df_lana[column].astype(str).value_counts().sort_index())
    print()

In [163]:
df_lana.to_csv('LLCP2023_clean.csv')

# PyCaret

In [ ]:
# check installed version
import pycaret
pycaret.__version__

'3.3.2'

In [ ]:
from pycaret.classification import *
# initialize setup
s = setup(data = df_lana, target = '_MENT14D', data_split_stratify=True, normalize = True, session_id=123)

In [ ]:
best = compare_models(sort = 'F1')

Processing:   0%|          | 0/69 [00:00<?, ?it/s]

In [ ]:
print(best)

GradientBoostingClassifier(ccp_alpha=0.0, criterion='friedman_mse', init=None,
                           learning_rate=0.1, loss='log_loss', max_depth=3,
                           max_features=None, max_leaf_nodes=None,
                           min_impurity_decrease=0.0, min_samples_leaf=1,
                           min_samples_split=2, min_weight_fraction_leaf=0.0,
                           n_estimators=100, n_iter_no_change=None,
                           random_state=123, subsample=1.0, tol=0.0001,
                           validation_fraction=0.1, verbose=0,
                           warm_start=False)


In [ ]:
model = create_model('gbc')

Processing:   0%|          | 0/4 [00:00<?, ?it/s]

In [ ]:
print(model)

GradientBoostingClassifier(ccp_alpha=0.0, criterion='friedman_mse', init=None,
                           learning_rate=0.1, loss='log_loss', max_depth=3,
                           max_features=None, max_leaf_nodes=None,
                           min_impurity_decrease=0.0, min_samples_leaf=1,
                           min_samples_split=2, min_weight_fraction_leaf=0.0,
                           n_estimators=100, n_iter_no_change=None,
                           random_state=123, subsample=1.0, tol=0.0001,
                           validation_fraction=0.1, verbose=0,
                           warm_start=False)


In [32]:
tuned_model = tune_model(model)

Initiated  . . . . . . . . . . . . . . . . . .                      17:46:26
Status     . . . . . . . . . . . . . . . . . .     Searching Hyperparameters
Estimator  . . . . . . . . . . . . . . . . . .  Gradient Boosting Classifier

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


KeyboardInterrupt: 

In [ ]:
print(tuned_model)

In [ ]:
evaluate_model(tuned_model)

In [ ]:
predict_model(tuned_model)

SEXVAR    GENHLTH POORHLTH EXERANY2 CVDSTRK3 CHCSCNC1 CHCOCNC1  \
35312     Male  Very good      1.0      Yes       No       No       No   
244829  Female       Good     None       No       No       No       No   
115408  Female       Good      7.0      Yes       No       No       No   
63755   Female  Very good     None      Yes       No       No       No   
371320  Female  Very good     None      Yes      Yes       No       No   
...        ...        ...      ...      ...      ...      ...      ...   
130652    Male  Very good     None      Yes       No       No       No   
360987    Male       Good     None       No       No       No       No   
248481    Male       Poor     25.0       No       No       No       No   
91291   Female  Very good     None      Yes       No       No       No   
81703     Male       Good     15.0      Yes       No       No       No   

       CHCCOPD3 ADDEPEV3 CHCKDNY2  ...      _AGEG5YR       _BMI5CAT  \
35312        No       No       No  ...  Age 50 to 54          Obese   
244829       No       No       No  ...  Age 60 to 64          Obese   
115408       No      Yes       No  ...  Age 40 to 44          Obese   
63755        No      Yes       No  ...  Age 40 to 44  Normal Weight   
371320       No       No       No  ...  Age 75 to 79     Overweight   
...         ...      ...      ...  ...           ...            ...   
130652       No       No       No  ...  Age 60 to 64     Overweight   
360987       No       No       No  ...  Age 55 to 59          Obese   
248481       No       No       No  ...       Age 80+  Normal Weight   
91291        No       No       No  ...  Age 50 to 54  Normal Weight   
81703        No      Yes       No  ...  Age 45 to 49     Overweight   

                 _CHLDCNT                      _EDUCAG _INCOMG1 _SMOKER3  \
35312                  No                  High School      4.0    Never   
244829                 No                  High School      5.0    Never   
115408                 No  College or Technical School      3.0    Never   
63755   Two or more child  College or Technical School      6.0    Never   
371320                 No  College or Technical School      3.0    Never   
...                   ...                          ...      ...      ...   
130652                 No             Attended College      6.0    Never   
360987                 No  College or Technical School      6.0    Never   
248481                 No  College or Technical School      5.0   Former   
91291                  No  College or Technical School      5.0   Former   
81703                  No                  High School      1.0      Yes   

       _RFDRHV8    _MENT14D prediction_label prediction_score  
35312        No   1-13 days        Zero days           0.7926  
244829       No   Zero days        Zero days           0.7214  
115408       No  \t14+ days        1-13 days           0.5824  
63755        No   1-13 days        1-13 days           0.7019  
371320       No   1-13 days        1-13 days           0.5235  
...         ...         ...              ...              ...  
130652       No   Zero days        Zero days           0.8869  
360987       No   1-13 days        Zero days           0.5904  
248481       No  \t14+ days        Zero days           0.4788  
91291        No   Zero days        1-13 days           0.6302  
81703        No   1-13 days       \t14+ days           0.9380  

[27515 rows x 30 columns]

In [ ]:
predict_model(tuned_model)

SEXVAR    GENHLTH POORHLTH EXERANY2 CVDSTRK3 CHCSCNC1 CHCOCNC1  \
35312     Male  Very good      1.0      Yes       No       No       No   
244829  Female       Good     None       No       No       No       No   
115408  Female       Good      7.0      Yes       No       No       No   
63755   Female  Very good     None      Yes       No       No       No   
371320  Female  Very good     None      Yes      Yes       No       No   
...        ...        ...      ...      ...      ...      ...      ...   
130652    Male  Very good     None      Yes       No       No       No   
360987    Male       Good     None       No       No       No       No   
248481    Male       Poor     25.0       No       No       No       No   
91291   Female  Very good     None      Yes       No       No       No   
81703     Male       Good     15.0      Yes       No       No       No   

       CHCCOPD3 ADDEPEV3 CHCKDNY2  ...      _AGEG5YR       _BMI5CAT  \
35312        No       No       No  ...  Age 50 to 54          Obese   
244829       No       No       No  ...  Age 60 to 64          Obese   
115408       No      Yes       No  ...  Age 40 to 44          Obese   
63755        No      Yes       No  ...  Age 40 to 44  Normal Weight   
371320       No       No       No  ...  Age 75 to 79     Overweight   
...         ...      ...      ...  ...           ...            ...   
130652       No       No       No  ...  Age 60 to 64     Overweight   
360987       No       No       No  ...  Age 55 to 59          Obese   
248481       No       No       No  ...       Age 80+  Normal Weight   
91291        No       No       No  ...  Age 50 to 54  Normal Weight   
81703        No      Yes       No  ...  Age 45 to 49     Overweight   

                 _CHLDCNT                      _EDUCAG _INCOMG1 _SMOKER3  \
35312                  No                  High School      4.0    Never   
244829                 No                  High School      5.0    Never   
115408                 No  College or Technical School      3.0    Never   
63755   Two or more child  College or Technical School      6.0    Never   
371320                 No  College or Technical School      3.0    Never   
...                   ...                          ...      ...      ...   
130652                 No             Attended College      6.0    Never   
360987                 No  College or Technical School      6.0    Never   
248481                 No  College or Technical School      5.0   Former   
91291                  No  College or Technical School      5.0   Former   
81703                  No                  High School      1.0      Yes   

       _RFDRHV8    _MENT14D prediction_label prediction_score  
35312        No   1-13 days        Zero days           0.8224  
244829       No   Zero days        Zero days           0.7176  
115408       No  \t14+ days        1-13 days           0.6479  
63755        No   1-13 days        1-13 days           0.6551  
371320       No   1-13 days        1-13 days           0.5395  
...         ...         ...              ...              ...  
130652       No   Zero days        Zero days           0.9044  
360987       No   1-13 days        Zero days           0.6045  
248481       No  \t14+ days        Zero days           0.4981  
91291        No   Zero days        1-13 days           0.6841  
81703        No   1-13 days       \t14+ days           0.9586  

[27515 rows x 30 columns]

In [ ]:
data_df.info()

In [ ]:
data_df.dtypes

In [ ]:
data_df.describe()

# PyCharet Install

In [ ]:
!pip install pycaret[full]

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.4/60.4 kB 4.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 165.8/165.8 kB 8.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 105.3/105.3 kB 8.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.3/44.3 kB 3.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 154.7/154.7 kB 10.6 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.0/4.0 MB 62.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.7/46.7 kB 3.6 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 50.1/50.1 kB 3.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.1/44.1 kB 3.1 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 177.5/177.5 kB 13.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━